pkg/proto/agentbaker/v1/config.proto

```diff
@@ -0,0 +1,228 @@
+syntax = "proto3";
+
+package agentbaker.v1;
+
+// illustrative division only
+message Configuration {
+  // System Configuration
+  
+  string cluster_certificate_authority = 1;
+
+  // only required until Secure TLS bootstrapping in place
+  // would use kubelet identity after that.
+  string tls_bootstrap_token = 2;
+
+  // cluster/user config
+  string kubernetes_version = 3; // Q: can this be auto-detected? Or is this part of specifying the desired node version?
+    // removed kubernetesHyperkubeSpec
+  string kube_binary_url = 4;      // maybe we can combine this and custom_kube_binary_url later once we figure out the logic of how they are used.
+  string custom_kube_binary_url = 5;
+  string kubeproxy_url = 6;
+  ApiServerConfig apiserver_config = 7;
+  string subscription_id = 8;
+  string resource_group = 9;
+  string location = 10;
+  string vm_type = 11;  
```

Comment:
IMDS?

Reply:
Will iterate on this in the upcoming PRs.